src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBinding.cs

```diff
@@ -38,7 +40,23 @@ public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
             if (value is SignalRTriggerEvent triggerEvent)
             {
                 var bindingContext = triggerEvent.Context;
-                // TODO: Add dynamic binding data in bindingData
+
+                // If ParameterNames are set, bind them in order.
+                // To reduce undefined situation, number of arguments should keep consist with that of ParameterNames
+                if (_attribute.ParameterNames != null && _attribute.ParameterNames.Length != 0)
+                {
+                    if (bindingContext.Arguments == null ||
+                        bindingContext.Arguments.Length != _attribute.ParameterNames.Length)
+                    {
+                        throw new SignalRTriggerParametersNotMatchException(_attribute.ParameterNames.Length, bindingContext.Arguments?.Length ?? 0);
+                    }
+
+                    var length = _attribute.ParameterNames.Length;
+                    for (var i = 0; i < length; i++)
+                    {
+                        bindingData.Add(_attribute.ParameterNames[i], bindingContext.Arguments[i]);
```

Comment:
>bindingData.Add(_attribute.ParameterNames[i], bindingContext.Arguments[i]); [](start = 24, length = 75)

Could this ensure binding order? The order of elements in a dictionary is non-deterministic, shall sort helps?

Reply:
The order in dictionary is insignificant. We can ensure the order in Invocation message's `Arguments` and that in Attribute's `ParameterNames` and they're both list.